framework/src/base/FEProblemBase.C

```diff
@@ -3662,6 +3663,13 @@ FEProblemBase::init()
 
   ghostGhostedBoundaries(); // We do this again right here in case new boundaries have been added
 
+  // do not assemble system matrix for JFNK solve
+  if (solverParams()._type == Moose::ST_JFNK)
+  {
+    _nl->system().set_basic_system_only();
+    _nl->nonlinearSolver()->jacobian = NULL;
+  }
```

Comment:
Nope, because _nl->init happens after _eq.init(), although we probably can move _nl->init above. That will require more changes which I am hesitating to do.